bot/resources/tags/blocking.md

```diff
@@ -0,0 +1,33 @@
+**What does blocking mean?**
+In asynchronous programming, blocking calls are all the parts of your function that are not using `await`. Not all forms of blocking are bad, and using blocking calls are inevitable, but make sure not to use too much, or else the program will freeze and you cannot do other tasks.
+
+**Examples of Blocking**
+A major example of blocking is using `time.sleep()`. Use `asyncio.sleep()` instead.
+Example:
+```py
+# bad
+time.sleep(10)
+# good
+await asyncio.sleep(10)
+```
+Another example is using the `requests` library. It's good for non-asynchronous programming, but certain requests can block the event too long with `asyncio`. Instead use `aiohttp`. Example:
+```py
+# bad
+r = requests.get('http://aws.random.cat/meow')
+if r.status_code == 200:
+    js = r.json()
+    await channel.send(js['file'])
+# good
+async with aiohttp.ClientSession() as session:
+    async with session.get('http://aws.random.cat/meow') as r:
+        if r.status == 200:
+            js = await r.json()
+            await channel.send(js['file'])
+```
+
+**How can I find asynchronous modules?**
+Most python modules have an asynchronous implementation. For example, `sqlite3` has `aiosqlite`, and `praw` has `asycpraw`. You try searching on [PyPi](https://pypi.org). Or check out [this list](https://github.com/timofurrer/awesome-asyncio).
```

Comment:
```suggestion
Most Python modules have an asynchronous implementation. For example, `sqlite3` has `aiosqlite`, and `praw` has `asycpraw`. You try searching on [PyPi](https://pypi.org), or check out [this curated list of Python asyncio frameworks](https://github.com/timofurrer/awesome-asyncio).
```
Grammar, and give the user info on what the link is to.

Reply:
Ok, good ideas!